target/linux/armvirt/image/Makefile

```diff
@@ -19,10 +19,23 @@ define Image/Build/Initramfs
 	)
 endef
 
+define Image/Build/gzip
+	gzip -f9n $(BIN_DIR)/$(IMG_PREFIX)-root.$(1)
+endef
+
+ifneq ($(CONFIG_TARGET_IMAGES_GZIP),)
+  define Image/Build/gzip/ext4
+	$(call Image/Build/gzip,ext4)
+  endef
+  define Image/Build/gzip/squashfs
+	$(call Image/Build/gzip,squashfs)
+  endef
+endif
+
 define Image/Build
 	$(call Image/Build/$(1))
-	dd if=$(KDIR)/root.$(1) bs=128k conv=sync | \
-		gzip -9n >$(BIN_DIR)/$(IMG_PREFIX)-root.$(1).gz
+	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-root.$(1) bs=128k conv=sync
```

Comment:
@jefferyto Do you need this dd line at all (same as in https://github.com/openwrt/openwrt/pull/1940#issuecomment-476141695)? The images should be produced by the previous step `$(call Image/Build/$(1))`

Reply:
As I mentioned in https://github.com/openwrt/openwrt/pull/1940#issuecomment-476148272, `dd` here is necessary, as `$(call Image/Build/$(1))` doesn't call anything.

Reply:
As you've explained already in #1940 it is needed, but I think, that we should use cp instead of the dd (and adjust the filenames to be consistent with x86), so I prefer following:

```
$(CP) $(KDIR)/root.$(1) $(BIN_DIR)/$(IMG_PREFIX)-rootfs-$(1).img
```

Reply:
Let's figure out malta before adjusting this.